packages/popper/src/index.vue

```diff
@@ -0,0 +1,466 @@
+  PositioningStrategy,
+} from '@popperjs/core'
+import useModifier from './useModifier'
+
+type Effect = 'dark' | 'light';
+
+const stop = (e) => e.stopPropagation()
+const throwError = (e: string) => {
+  throw new Error(e)
+}
+const getTrigger = () => {
+  const {
+    subTree: { dynamicChildren },
+  } = getCurrentInstance()
+  return dynamicChildren[0].children
+}
+
+export default defineComponent({
+  name: 'ElPopper',
+  props: {
+    arrowOffset: {
+      type: Number,
+      default: 15,
+    },
+    appendToBody: {
+      type: Boolean,
+      default: true,
+    },
+    boundariesPadding: {
+      type: Number,
+      default: 0,
+    },
+    content: {
+      type: String,
+      default: '',
+    },
+    closeDelay: {
+      type: Number,
+      default: 200,
+    },
+    cutoff: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      defualt: false,
+    },
+    effect: {
+      type: String as PropType<Effect>,
+      default: 'dark' as Effect,
+    },
+    enterable: {
+      type: Boolean,
+      default: true,
+    },
+    flip: {
+      type: Boolean,
+      default: true,
+    },
+    hideAfter: {
+      type: Number,
+      default: 0,
+    },
+    manualMode: {
+      type: Boolean,
+      default: false,
+    },
+    openDelay: {
+      type: Number,
+      default: 0,
+    },
+    offset: {
+      type: [Number, Array] as PropType<[number, number] | number>,
+      default: [0, 12] as [number, number],
+      validator: (val: [number, number] | number): boolean => {
+        return (isArray(val) && val.length === 2) || typeof val === 'number'
+      },
+    },
+    placement: {
+      type: String as PropType<Placement>,
+      default: 'bottom' as Placement,
+    },
+    popperClass: {
+      type: String,
+      default: '',
+    },
+    referrer: {
+      type: HTMLElement as PropType<Nullable<HTMLElement>>,
+      default: null as Nullable<HTMLElement>,
+    },
+    showArrow: {
+      type: Boolean,
+      default: true,
+    },
+    strategy: {
+      type: String as PropType<PositioningStrategy>,
+      default: 'fixed' as PositioningStrategy,
+    },
+    transition: {
+      type: String,
+      default: 'el-fade-in-linear',
+    },
+    tabIndex: {
+      type: String,
+      default: '0',
+    },
+    value: {
+      type: Boolean,
+      default: true,
+    },
+  },
+  setup(props, { slots }) {
+    const popperRef = ref(null as Nullable<HTMLElement>)
+    const arrowRef = ref(null as Nullable<HTMLElement>)
+    const popperInstance = ref(null as PopperInstance)
+    const exceptionState = ref(false)
+    const timeout = ref(null as NodeJS.Timeout)
+    const timeoutPending = ref(null as NodeJS.Timeout)
+    const show = ref(false)
+    const popperId = ref(`el-tooltip-${generateId()}`)
+    const trigger = ref(null as Nullable<HTMLElement>)
+
+    const visible = computed(() => {
+      return props.manualMode ? props.value : !props.disabled && show.value
```

Comment:
why not `modelValue`?

Reply:
Actually, this one is a underlaying component for components like `tooltip` `popover`, so I don't think that we need to implement the exact same API as `tooltip` did